### PR TITLE
Fix the portal Play sidebar and stop HTML markup leaking to plain telnet

### PR DESCRIPTION
**Pull request by HarryCordewener**
Three defects behind "the Play sidebar is empty and exits print HTML". Each was reproduced against a live server (Server + ConnectionServer + NATS, real telnet and real browser) before being fixed.

## 1. `Here` / `Exits` never populated

The `ROOM`CONTENTS` handler that emits the `room.contents` / `room.exits` OOB packages existed **only as documentation** (`docs/softcode/room-contents-handler.md`, which said "install the handler manually; packaging is a follow-up"). Nothing installed it, so a stock server emitted no OOB frames at all and `Play.razor`'s sidebar had no data source. The engine side was already fine — the event fires room-scoped on move/connect/disconnect, `oob()` delivers, and the client's channel store works.

Evidence: zero OOB frames on a fresh game; correct payloads the moment the handler was pasted in by hand.

Fix — ship it as a bundled package:

- `examples/packages/room-contents/` — attach mode on `{{$event_handler}}`, helpers rewritten `me/`-relative instead of hardcoding `#9`
- added `event_handler` to `WellKnownRefs` and resolved it in `PackageInstallService.BuildWellKnownMapAsync` (only `http_handler` was resolvable before)
- registered in `BundledPackages.All` + embedded in the Server csproj; the attach gate is now per-handler (`BundledPackageHandler.Http/Event/None`) rather than a `RequiresHttpHandler` bool

## 2. Bundled packages had no upgrade path

`DefaultPackagesBootstrapService.InstallIfAbsentAsync` returned early for *any* already-installed package regardless of version. So a game kept whatever version of a bundled package it first installed, forever.

Concretely: `GET`ONLINE` / `FN`ONLINEVIS` only arrived in profile-handler **1.1.0** (be4c8dea). A game whose DB predates that keeps 1.0.0 → `/http/online` 404s → `CharacterDirectoryService` swallows the `HttpRequestException` and the "Online Now" widget renders its empty state, indistinguishable from genuinely nobody being connected.

Fix: bootstrap upgrades in place when the build ships a strictly newer version, reusing the existing `KeepMine` conflict resolution so local edits survive. Same-or-newer installed is still left to the package manager.

## 3. `<send>` tags reached non-Pueblo/non-MXP clients

`MarkupOutputRenderer.Render` used `ms.ToString()` for the ANSI case. That is not the ANSI render — it is `NativeToString()`, which renders **every markup span in its own native form**: `AnsiMarkup` → ANSI codes (fine), `HtmlMarkup` → the literal tag. Since `look` wraps every exit name in `WrapExitInSendTag`, a plain telnet client got:

```
Obvious exits:
<send href="north" hint="Go north">north</send>
```

Fix: use the ANSI render, which maps `HtmlMarkup` to its ANSI equivalent (`b`/`i`/`u`/`s`) or to bare text. Same client after the fix prints `north`; an MXP-negotiating client on the same build still gets its `<send>` tags behind the `ESC[0z` prefix.

This also covers `tagwrap()` in softcode (the other `HtmlMarkup` producer) and prompts, which share the renderer. For ANSI-only content the change is a no-op — the two strategies are identical when the first markup is an `AnsiMarkup`. Where they differ the ANSI render is also strictly better: it appends the standard trailing reset that the native path skipped whenever an `HtmlMarkup` came first, so a coloured exits line could previously bleed colour past its end.

## Why it wasn't caught

`PuebloMxpRenderTests` asserts `Render("ansi")` strips `<send>` — correct, but that tests the `MString` API, not the path the ConnectionServer takes. `MarkupOutputRendererTests.Ansi_KeepsRawText` fed it a plain string that merely *contains* the characters `<send …>`, with no markup runs. Nothing exercised a real `HtmlMarkup` run through the renderer.

## Tests added

- `MarkupOutputRendererTests` — a bare send span, the mixed `"Obvious exits:\n"` + send line as `look` builds it (the leading plain run is what makes the `HtmlMarkup` the *first* markup, which selected the native strategy), and a colour+send span asserting ANSI survives
- `BundledPackagesTests` — every entry in `BundledPackages.All` has an embedded manifest that parses, and attach-mode packages declare the handler they target (a missing `EmbeddedResource` was silent-at-boot)
- `BundledPackageUpgradeTests` — the version gate, including unparseable/missing versions
- `RoomContentsPackageTests` (integration) — the package installs at boot and its attributes land on the *configured* event handler
- `BundledPackageUpgradeIntegrationTests` — re-applying a newer manifest of an installed package really does add the new routes and move the recorded version forward

## Verification

Locally on this branch: unit **4690 passed / 198 skipped / 0 failed**, integration **246/246**, BUnit **263/263** (ArangoDB provider). Plus live checks: plain telnet shows `north`, MXP shows the tag, and the portal renders **Here** → God, Package Manager · **Exits** → north · **Online Now** → God with no manual softcode.

## Note on the docs

`docs/softcode/room-contents-handler.md` is rewritten to describe an installed package rather than a manual procedure, and its "the real path runs the handler with God (#1) as the executor" claim is corrected — `EventService` sets `Executor: handlerRef`, i.e. the handler object itself (`me`), which is why the packaged handler can use `me/FN`WHOVIS`.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Added the bundled `room-contents` package to keep the web portal sidebar updated with room occupants and exits.
  * Bundled packages can now attach via event-based handlers and resolve `event_handler` wiring.
* **Bug Fixes**
  * ANSI output now strips HTML `<send>` markup correctly while preserving ANSI styling.
  * Configured handler resolution no longer falls back to guessing when handlers are missing.
  * Bundled package upgrades now apply only for strictly newer versions, preserving local/admin customizations.
* **Documentation**
  * Updated extensibility overview and refreshed `room-contents` package/handler documentation.

### CODE REVIEW

File: SharpMUSH.Server/Services/BundledPackages.cs [hunk 1]
```diff
@@ -9,22 +9,37 @@ namespace SharpMUSH.Server.Services;
 /// each entry here is a bundled <c>examples/packages/&lt;id&gt;/package.yaml</c> embedded as a
 /// resource. Adding a default package is a one-line addition to <see cref="All"/>.
 /// </summary>
+/// <summary>The configured handler object a bundled attach-mode package needs.</summary>
+public enum BundledPackageHandler
```
[the remark this answers is not in the document]
Comment (HarryCordewener) [reply SharpMUSH.Server/Services/BundledPackages.cs]: Fixed in e72dc8c9 — the enum now sits above the class doc, so `BundledPackages` keeps its summary and the enum has exactly one.

File: examples/packages/room-contents/README.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  the `goto` command a client issues to traverse it.
+
+The portal routes incoming OOB frames by package name into its per-connection
+channel store; the Play sidebar renders the `who` / `exits` entries and issues
+an entry's `cmd` on click. Without a handler attribute the engine emits nothing
+and the sidebar stays empty.
+
+It manages only these attributes on the configured `event_handler` object
+(`{{$event_handler}}`, #9) — it never creates or destroys the object, and
+uninstalling leaves the handler object's other softcode untouched.
```
[the remark this answers is not in the document]
Comment (HarryCordewener) [reply examples/packages/room-contents/README.md]: Fixed in e72dc8c9 — now reads "`#9` by default" and states the target is resolved from config at install time.

File: docs/softcode/room-contents-handler.md [hunk 3]
```diff
@@ -1,18 +1,18 @@
...
-
-> **Packaging note:** The eventual home for this softcode is a SharpMUSH
-> package loadable via the `@package` command. Until the package-manager
-> integration ships as a follow-up, install the handler manually as shown
-> below.
+# WebSocket Support Package — the `ROOM`CONTENTS` Handler
+
+This document is part of the **WebSocket Support Package**. It describes the
+`ROOM`CONTENTS` event handler that fans out structured OOB pushes to a room's
+connected occupants whenever the room's population changes (player movement,
```
[the remark this answers is not in the document]
Comment (HarryCordewener) [reply docs/softcode/room-contents-handler.md]: Confirmed and fixed in e72dc8c9. A span opened with one backtick terminates at the next backtick, so ``ROOM`CONTENTS`` rendered as the span "ROOM" followed by loose text. Converted the affected prose spans in this doc to double-backtick form (fenced blocks were already fine). That also surfaced a stale reference: idiom 5 still named the removed `FN`NOTEXIT` helper rather than `FN`WHOVIS`, so it is updated too.

File: SharpMUSH.Server/Services/DefaultPackagesBootstrapService.cs [hunk 4]
```diff
@@ -37,31 +39,55 @@ public async Task StartAsync(CancellationToken cancellationToken)
...
+			await InstallOrUpgradeAsync(package.PackageId, cancellationToken);
 		}
 	}
 
-	private async Task InstallIfAbsentAsync(string packageId, CancellationToken cancellationToken)
+	private async Task InstallOrUpgradeAsync(string packageId, CancellationToken cancellationToken)
 	{
+		var parsedForVersion = manifests.ParseManifest(BundledPackages.ManifestYaml(packageId));
+		var bundledVersion = parsedForVersion.IsT0 ? parsedForVersion.AsT0.Manifest.Version : null;
+
```
[the remark this answers is not in the document]
Comment (HarryCordewener) [reply SharpMUSH.Server/Services/DefaultPackagesBootstrapService.cs]: Confirmed and fixed in aad19c48. Worse than cosmetic: an unparsed manifest yields a null version, which fails the `IsNewer` gate, so on any game that already had the package the build defect was logged at Debug as routine ("already installed") rather than as an error — it was only ever surfaced on games that did not have it. Parsing and validating now happen before the version gate, and the gate reads the version off the parsed manifest, which also drops the redundant local alias.

File: docs/softcode/room-contents-handler.md [hunk 5]
```diff
@@ -116,30 +117,39 @@ Verify it is set:
 
 ---
 
-## Removing the handler
+## Silencing the handler
 
 ```
 &ROOM`CONTENTS #9=
 ```
```
[the remark this answers is not in the document]
Comment (HarryCordewener) [reply docs/softcode/room-contents-handler.md]: Applied in dbe80fa9 — the three bare fences in this doc are now ```text. Scoped to this file, which already labels its mushcode and json fences; the repo has ~670 unlabelled fences elsewhere and no markdownlint config or CI job, so a wider sweep would be inconsistent noise here and belongs in its own change.

File: SharpMUSH.Tests.Integration/Packages/BundledPackageUpgradeIntegrationTests.cs [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+			// when it gained GET`ONLINE, and what a game installed at 1.0.0 never received.
+			await ApplyAsync(Manifest("1.1.0", "      PROBE`ADDED: |-\n        second"));
+
+			var upgraded = await Registry.GetInstalledPackageAsync(PackageId);
+			await Assert.That(upgraded.IsT0).IsTrue();
+			await Assert.That(upgraded.AsT0.Version).IsEqualTo("1.1.0");
+
+			var managed = (await Registry.GetManagedAttributesAsync(PackageId)).Select(m => m.Attribute).ToList();
+			await Assert.That(managed).Contains("PROBE`ADDED");
+			await Assert.That(managed).Contains("PROBE`ORIGINAL");
```
[the remark this answers is not in the document]
Comment (HarryCordewener) [reply SharpMUSH.Tests.Integration/Packages/BundledPackageUpgradeIntegrationTests.cs]: Correct, and it undercut the test's own summary — it claimed local edits survive but created no drift, so nothing ever reached KeepMine. Fixed in dbe80fa9: the test now edits PROBE`ORIGINAL in-game between installs and ships a different value for it in v1.1.0, so the upgrade produces both an Add (PROBE`ADDED) and a ModifyModify conflict, then asserts the admin's value survived and the new attribute went live. ApplyAsync also returns how many conflicts it resolved and the test asserts that is non-zero, so it cannot pass by no conflict arising.

File: examples/packages/room-contents/README.md [hunk 7]
```diff
@@ -0,0 +1,36 @@
+# room-contents
+
+The `` ROOM`CONTENTS `` event handler, delivered as an attach-mode package
```
[the remark this answers is not in the document]
Comment (HarryCordewener) [reply examples/packages/room-contents/README.md:3]: Not changing this one — the premise is incorrect. CommonMark strips one leading and one trailing space from a code span when both are present, so the two forms are byte-identical after parsing. Checked with a CommonMark parser:

```
`` ROOM`CONTENTS ``  ->  <p><code>ROOM`CONTENTS</code></p>
``ROOM`CONTENTS``    ->  <p><code>ROOM`CONTENTS</code></p>
```

The spaced form is also the established convention for backtick-containing attribute names in this repo, in the sibling package READMEs this file sits next to:

- `examples/packages/http-handler/README.md:7` — `` GET`FOO`BAR ``
- `examples/packages/profile-handler/README.md:8-9` — `` GET`CHARACTERS ``, `` GET`PROFILE ``
- `examples/packages/README.md:126-137` — `` PM`REFS`NAME ``, `` PM` ``

So switching would move away from the convention rather than toward it. Happy to normalise if you would rather standardise on the unspaced form, but that is a repo-wide docs decision, not this PR.

File: examples/packages/room-contents/README.md [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+
+It manages only these attributes on the configured `event_handler` object
+(`{{$event_handler}}`, `#9` by default) — the target is resolved from config at
+install time, never a literal dbref. It never creates or destroys the object,
+and uninstalling leaves the handler object's other softcode untouched.
+
+## Customising
+
+The helpers are the intended seam: `` FN`WHOVIS `` decides who is listed (add
+dark/visibility guards here), and `` FN`WHOROW `` / `` FN`EXITROW `` build one row
```
[the remark this answers is not in the document]
Comment (HarryCordewener) [reply examples/packages/room-contents/README.md:31]: Not changing this one — the premise is incorrect. CommonMark strips one leading and one trailing space from a code span when both are present, so the two forms are byte-identical after parsing. Checked with a CommonMark parser:

```
`` ROOM`CONTENTS ``  ->  <p><code>ROOM`CONTENTS</code></p>
``ROOM`CONTENTS``    ->  <p><code>ROOM`CONTENTS</code></p>
```

The spaced form is also the established convention for backtick-containing attribute names in this repo, in the sibling package READMEs this file sits next to:

- `examples/packages/http-handler/README.md:7` — `` GET`FOO`BAR ``
- `examples/packages/profile-handler/README.md:8-9` — `` GET`CHARACTERS ``, `` GET`PROFILE ``
- `examples/packages/README.md:126-137` — `` PM`REFS`NAME ``, `` PM` ``

So switching would move away from the convention rather than toward it. Happy to normalise if you would rather standardise on the unspaced form, but that is a repo-wide docs decision, not this PR.

File: docs/softcode/room-contents-handler.md [hunk 9]
```diff
@@ -77,69 +77,79 @@ These were learned the hard way; the reference handler relies on all of them:
...
 
 ---
 
-## Removing the handler
+## Silencing the handler
 
-```
+```text
 &ROOM`CONTENTS #9=
 ```
```
[the remark this answers is not in the document]
Comment (HarryCordewener) [reply docs/softcode/room-contents-handler.md:128]: Valid concern, addressed differently in fa982ec3. Substituting a placeholder into just these two commands would leave the mushcode block directly above them still using `#9`, so the section would be half-placeholder and half-literal. Instead the section now states once, before all the examples, that they target `#9` as the default `event_handler` and that a game configuring a different handler should substitute its dbref throughout — which also covers the install block your diff did not touch.

Two notes on the suggestion itself: the committable block came through empty, and the diff had mangled backticks (`> get `#9/ROOM``CONTENTS`), so it was not applicable as-is.

File: SharpMUSH.Tests/Packages/BundledPackagesTests.cs [hunk 10]
```diff
@@ -0,0 +1,86 @@
...
+	/// An attach-mode bundled package must declare the handler it attaches to, or bootstrap
+	/// would try to install it on a game that has no target for its {{$...}} target ref.
+	/// </summary>
+	[Test]
+	public async Task AttachModePackages_DeclareTheHandlerTheyTarget()
+	{
+		foreach (var descriptor in BundledPackages.All)
+		{
+			var manifest = _manifests.ParseManifest(BundledPackages.ManifestYaml(descriptor.PackageId)).AsT0.Manifest;
+
```
[the remark this answers is not in the document]
Comment (HarryCordewener) [reply SharpMUSH.Tests/Packages/BundledPackagesTests.cs]: Fixed in bd202dbc. It was also inconsistent with the other two tests in the same file, which do assert the parse before unwrapping. The test now asserts `IsT0` with the package id in the message, so it fails readably on its own rather than throwing an opaque OneOf exception and leaning on `EveryBundledPackage_HasAnEmbeddedManifestThatParses` having run first.
